docs/blog/posts/release1.0.0.md

```diff
@@ -0,0 +1,276 @@
+
+## Spatial map
+
+Textual apps typically contain many widgets of different sizes and at different locations within the terminal.
+Not all of which widgets may be visible in the final view (if they are within a scrolling container).
+
+
+!!! info "The smallest Widget"
+
+    While it is possible to have a widget as small as a single character, I've never found a need for one.
+    The closest we get in Textual is a [scrollbar corner](https://textual.textualize.io/api/scrollbar/#textual.scrollbar.ScrollBarCorner);
+    a widget which exists to fill the space made when a vertical scrollbar and a horizontal scrollbar meet.
+    It does nothing because it doesn't need to, but it is powered by an async task like all widgets and can receive input.
+    I have often wondered if there could be something useful in there.
+    A game perhaps?
+    If you can think of a game that can be played in 2 characters &mdash; let me know!
+
+The *spatial map*[^1] is a data structure used by the compositor to very quickly discard widgets that are not visible within a given region.
+The algorithm it uses may be familiar if you have done any classic game-dev.
+
+
+### The problem 
+
+Consider the following arrangement of widgets:
+
+<div class="excalidraw">
+--8<-- "docs/blog/images/compositor/spatial-map.excalidraw.svg"
+</div>
+
+Here we have 8 widgets, where only around 4 will be visible at any given time, depending on the position of the scrollbar.
+We want to avoid doing work on widgets which will not be seen in the next frame.
+
+A naive solution to this would be to check each widget's [Region][textual.geometry.Region] to see if it overlaps with the visible area.
+This is a perfectly reasonable solution, but it won't scale well.
+If we get in to the 1000s of widgets territory, it may become significant &mdash; and we may have to do this 30 times a second if we are scrolling.
+
+### The Grid
+
+The first step in the spatial map is associate every widget with a tile in a regular grid[^2].
+
+<div class="excalidraw">
+--8<-- "docs/blog/images/compositor/spatial-map-grid.excalidraw.svg"
+</div>
+
+The size of the grid is fairly arbitrary, but it should be large enough to cover the viewable area with a relatively small number of grid tiles.
+We use a grid size of 100 characters by 20 lines, which seems about right.
+
+When the spatial map is first created it places each widget in one or more grid tiles.
+At the end of that process we have a dict that maps every grid coordinate on to a list of widgets, which will look something like the following:
+
+```python
+{
+    (0, 0): [widget1, widget2, widget3],
+    (1, 0): [widget1, widget2, widget3],
+    (0, 1): [widget4, widget5, widget6],
+    (1, 1): [widget4, widget5, widget6],
+    (0, 2): [widget7, widget8],
+    (1, 2): [Widget7, widget8]
+}
+```
+
+This data is cacheable.
+If the widgets don't change their position or size such as when user is *scrolling*, then we can reuse the information.
+
```

Comment:
Does this mean we can reuse the information while scrolling or we can't? I can read it both ways

Reply:
New version is great, cheers 👍